sdk/go/common/resource/plugin/plugin.go

```diff
@@ -530,6 +539,46 @@ func execPlugin(ctx *Context, bin, prefix string, kind apitype.PluginKind,
 	}, nil
 }
 
+func inferRuntime(pluginDir string) (workspace.ProjectRuntimeInfo, error) {
+	// Define the file-to-runtime mapping
+	runtimeFiles := map[string]string{
+		"package.json":     "nodejs",
+		"requirements.txt": "python",
```

Comment:
What about poetry and uv?

Reply:
I think we don't support those for authoring components right now so that not to require the consumers to have poetry/uv installed. We may add those later.